### DISCUSSION TITLE
Settings for the private OpenAI server

### DISCUSSION THREAD

**Discussion by iG8R** [immersive-translate/immersive-translate#904] [Help] [answered]
### 概述 | Summary

In my company there is a private translation server which supports different translation services and among them the most important one for me is ChatGPT.
Is there any way to setup OpenAI settings so as I could be able to use it via ImmersiveTranslate?
Below is an example of a request to the server API.

![ImmersiveTranslate-ownChatGPTserver-Headers03-2023-09-04_200146](https://github.com/immersive-translate/immersive-translate/assets/11407417/13093e02-4b7e-4950-84fa-c13999000da5)
![ImmersiveTranslate-ownChatGPTserver-Request-2023-09-04_195719](https://github.com/immersive-translate/immersive-translate/assets/11407417/18cfcdeb-25be-4b7c-857e-ef69e93fe48d)
![ImmersiveTranslate-ownChatGPTserver-Response02-2023-09-04_195745](https://github.com/immersive-translate/immersive-translate/assets/11407417/5a1ba4e0-aefb-477b-be91-b7f9393057ab)

**Comment by theowenyoung** [comment, marked as answer]
No, you should design your custom interface in the format of OpenAI's request and response.

In theory, you could run a simple forwarding service locally, and your forwarding service would support OpenAI's standard inputs and outputs, and then forward requests to your company's private server

**Reply by iG8R** [reply]
Thanks a lot.

**Reply by iG8R** [reply]
One more question.
In theory, is it possible to implement something like "plugin support", which would allow users to connect new translation services to ImmersiveTranslate without interfering with its code itself, or perhaps implement something like a "request constructor" within ImmersiveTranslate itself, which would allow users to form the necessary translation service requests?
